### ISSUE TITLE
support for iOS devices with iOS 10?

### ISSUE THREAD

**Issue by Bart274** [schollz/find#108]
Hi,

Do you think it's possible to get this working with iOS 10 now?

**Comment by schollz** [comment]
Good news, apparently [it is available in iOS 9+](http://stackoverflow.com/questions/32970711/is-it-possible-to-get-wifi-signal-strength-in-ios-9/32971064#32971064).

I would need help to develop this since I've never programmed for iOS devices.

**Comment by Bart274** [comment]
I'm glad to see it should work for iOS devices now as well! :)
Unfortunately, I can't help with the development as I have no experience for developing for iOS :(

Op 2 aug. 2016 om 16:43 heeft Zack <notifications@github.com<mailto:notifications@github.com>> het volgende geschreven:


Good news, apparently it is available in iOS 9+<http://stackoverflow.com/questions/32970711/is-it-possible-to-get-wifi-signal-strength-in-ios-9/32971064#32971064>.

I would need help to develop this since I've never programmed for iOS devices.

-

**Comment by abcdev** [comment]
I have requested a hotspot helper network extension in early august but have received response.

**Comment by schollz** [comment]
Do you know how long it usually takes?

**Comment by abcdev** [comment]
Nope, but my request dates back almost three weeks. It seems quite long even for apple standards. Maybe they don't reply when they reject a request ?

**Comment by jonathanweinberg** [comment]
I'm not shocked that are running wickedly behind since there is steady rumors of not only the iPhone refresh, but several other platforms being refreshed alongside (MacBook\iMac + Watch).

Wish there were some way to help.

**Comment by Bart274** [comment]
I really hope someone can help us out with this :)
that, or we can use bluetooth?

**Comment by abcdev** [comment]
This is the response from apple I got today:

Hello,

Thank you for your interest in the Network Extension framework. Unfortunately, these APIs are not designed for the use you’ve identified. We’ll let you know if that changes.

**Comment by schollz** [comment]
That's really unfortunate. I wonder if "not designed" means "not allowed"?

**Comment by jonathanweinberg** [comment]
And along that same token, if you happened to do one of the "intended" purposes, while doing the things we need to do (fingerprinting sequence), it might be allowable.

**Comment by jonathanweinberg** [comment]
Maybe after launch of iOS 11, another request might be possible?

**Comment by schollz** [comment]
That would be fantastic. I'm not an iOS person, so if someone else can weigh in that would be helpful!

**Comment by abcdev** [comment]
@schollz By now I have played around with the NEHotspotHelper API. Bad news though, it is not feasible for an indoor location service. You get no guarantee to be called when having registered a Hotspot Helper. It is triggered every time when opening the WiFi settings but thats the only "pattern" I saw. It is designed to help the user choose a WiFi an app provides further info for but that's all. See also [Apple Developer Forum](https://forums.developer.apple.com/thread/47300)
